bindings_generator/src/methods.rs

```diff
@@ -348,10 +348,15 @@ pub(crate) fn generate_methods(
 
         let rusty_name = format_ident!("{}", rusty_method_name);
 
-        let maybe_unsafe = if UNSAFE_OBJECT_METHODS.contains(&(&class.name, method_name)) {
-            quote! { unsafe }
+        let maybe_unsafe: TokenStream;
+        let maybe_unsafe_reason: &str;
+        if UNSAFE_OBJECT_METHODS.contains(&(&class.name, method_name)) {
+            maybe_unsafe = quote! { unsafe };
+            maybe_unsafe_reason = "**Safety:** This function bypasses Rust's static type checks \
+                (aliasing, thread boundaries, calls to free(), ...).";
```

Comment:
Safety documentation is usually formatted as Markdown sections with a `h1` header, and ordered after the method descrtiption. Example: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.from_raw_parts

Reply:
Good point, fixed